lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,1079 @@
+    - Etienne Desautels (@etienned)
+short_description: Manipulates macOS preferences including complex data types
+version_added: "2.6"
+description:
+    - This module allows users to create, read, update, and delete system
+      and application preferences on macOS including deeply nested dictionary
+      values like those in the C(com.apple.finder) domain.
+    - It also provides a convenient C(merge) strategy for assignments in
+      nested dictionaries so that users can very specifically target nested
+      keys without having to assign unchanged peer or parent values.
+requirements:
+    - Target machine should be running macOS
+    - PyObjc python module (preinstalled when using system python since macOS 10.5 Leopard).
+      When using python 3 you have to install PyObjc manually.
+options:
+    domain:
+        description:
+            - The preference domain. E.g. C(com.apple.finder), C(/path/to/some.plist), C(NSGlobalDomain).
+        required: false
+        default: NSGlobalDomain
+    user:
+        description:
+            - The user on which the preference will be applied.
+              C(anyUser) will set the preference for all users on the machine.
+            - See U(https://developer.apple.com/library/content/documentation/CoreFoundation/Conceptual/CFPreferences/Concepts/PreferenceDomains.html)
+              for further information.
+        required: false
+        default: currentUser
+        choices: ["anyUser", "currentUser"]
+    host:
+        description:
+            - The host on which the preference should apply. Most Apple preferences
+              live in the C(anyHost) domain.
+            - See U(https://developer.apple.com/library/content/documentation/CoreFoundation/Conceptual/CFPreferences/Concepts/PreferenceDomains.html)
+              for further information.
+        required: false
+        default: anyHost
+        choices: ["anyHost", "currentHost"]
+    key:
+        description:
+            - The key of the preference. Nested values can be accessed by giving all
+              keys and indexes separated by colons (:). Indexes are zero-based
+              (see Notes for special cases).
+        required: true
+    type:
+        description:
+            - The type of the value to write. If unspecified, I(type) will
+              be deduced mostly the same way YAML casts types (see Notes for special cases).
+        required: false
+        default: null
+        choices: ["array", "bool", "boolean", "data", "date", "dict", "float", "real", "int", "integer", "string"]
+    value:
+        description:
+            - The value that will be set for the specified key.
+              If C(value) is omitted and C(state) is not C(absent), the preference value of C(key) will be returned.
+        required: false
+    state:
+        description:
+            - The state of the preference.
+              c(merge) performs a deep merge of dictionaries and arrays.
+        required: false
```

Comment:
If it's not required, you don't need to add this.
Besides, it has a default value, so in a way it is required ;-)